library/cartridge_connect_to_membership.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+
+import math
+import pkgutil
+
+if pkgutil.find_loader('ansible.module_utils.helpers'):
+    import ansible.module_utils.helpers as helpers
+else:
+    import module_utils.helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+    'hostvars': {'required': True, 'type': 'dict'},
+    'play_hosts': {'required': True, 'type': 'list'},
+}
+
+BATCH_SIZE = 32
+
+
+def connect_instances(control_console, play_hosts, instance_names, advertise_uris):
+    connect_errors, _ = control_console.eval_res_err('''
+        local advertise_uris = ...
+
+        local cartridge = require('cartridge')
+        local fiber = require('fiber')
+
+        local count = 0
+        local channel = fiber.channel()
+
+        for _, advertise_uri in ipairs(advertise_uris) do
+            count = count + 1
+
+            fiber.create(function()
+                local _, err = cartridge.admin_probe_server(advertise_uri)
+                channel:put({
+                    advertise_uri = advertise_uri,
+                    err = err and err.err or err,
+                })
+            end)
+        end
+
+        local errors = {}
+        for _ = 1, count do
+            local res = channel:get(60)
```

Comment:
Also it's quite strange to use 60 for each call. In fact you have N * 60 seconds.

You need something like:
```
local deadline = fiber.clock() + timeout

for ... do
    local get_timeout = deadline - fiber.clock()
    if get_timeout < 0 then
        get_timeout = 0
    end
    get(get_timeout)
end
```